ui/app/components/secure-variable-form.js

```diff
@@ -31,6 +32,23 @@ export default class SecureVariableFormComponent extends Component {
    * @type {DuplicatePathWarning}
    */
   @tracked duplicatePathWarning = null;
+  @tracked variableNamespace = null;
+  @tracked namespaceOptions = null;
+
+  @action
+  setNamespace(namespace) {
+    this.variableNamespace = namespace;
+  }
```

Comment:
Suggestion: this should inherit from the `?namespace=...` query param that gets set by selecting a namespace on the path/index page before creation. That query param persists to the /new page, so we can use it to help save the user a click.

Reply:
Yes, good point. But, this action is about setting a property on a form, this isn't being used to filter, sort and be passed to a request and its really about form state so I wouldn't have it be a query parameter because then it would imply one of the cases above.

The query param shouldn't persist to the `/new` route.